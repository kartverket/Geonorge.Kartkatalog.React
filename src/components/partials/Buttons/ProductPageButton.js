import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import style from './Buttons.scss';

import { getResource } from '../../../actions/ResourceActions'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export class ProductPageButton extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    render() {
        let buttonDescription = this.props.getResource('DisplayProductPage', 'Vis produktside');
        // TODO styling
        if (this.props.metadata.ProductPageUrl) {
            let url = this.props.metadata.ProductPageUrl
            let icon = <FontAwesomeIcon title={buttonDescription} icon={['far', 'external-link-square']} key="icon" />;
            let buttonClass = style.btn;
            let textContent = React.createElement('span', { key: "textContent" }, buttonDescription);

            let childElements = [icon, textContent];
            return React.createElement('a', { href: url, className: buttonClass }, childElements);
        } else {
            let icon = <FontAwesomeIcon title={buttonDescription} icon={['far', 'external-link-square']} key="icon" />
            let buttonClass = style.btn + ' disabled'
            let textContent = React.createElement('span', { key: "textContent" }, buttonDescription);
            let childElements = [icon, textContent];
            return React.createElement('span', { className: buttonClass }, childElements);
        }
    }

}

ProductPageButton.propTypes = {
    metadata: PropTypes.object.isRequired
};

const mapDispatchToProps = {
    getResource
};

const mapStateToProps = state => ({
    resources: state.resources
});

export default connect(mapStateToProps, mapDispatchToProps)(ProductPageButton);
