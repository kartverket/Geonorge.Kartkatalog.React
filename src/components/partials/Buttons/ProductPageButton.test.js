import React from 'react';
import { shallow } from "enzyme";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { ProductPageButton } from './ProductPageButton';
import style from 'components/partials/Buttons/Buttons.module.scss'

function setupItemWithProductPageUrl() {
    const metadata = {
        Uuid: '1234',
        ProductPageUrl: 'test'
    }
    const props = {
        metadata: metadata,
        getResource: jest.fn()
    }
    const wrapper = shallow(<ProductPageButton {...props} />)

    return {
        props,
        wrapper
    }
}

function setupItemWithoutProductPageUrl() {
    const metadata = {
        Uuid: '1234',
    }
    const props = {
        metadata: metadata,
        getResource: jest.fn()
    }
    const wrapper = shallow(<ProductPageButton {...props} />)

    return {
        props,
        wrapper
    }
}


describe('setupItemWithProductPageUrl', () => {

    it('should render self', () => {
        const { wrapper } = setupItemWithProductPageUrl()
        expect(wrapper).toMatchSnapshot();
    })

    it('Item with product page url', () => {
        const { wrapper } = setupItemWithProductPageUrl()

        expect(wrapper.hasClass(style.btn)).toBe(true)
        expect(wrapper.prop("href")).toBe('test')
        expect(wrapper.find(FontAwesomeIcon).first().prop("icon")).toContain('external-link-square')
    })

    it('Item without product page url', () => {
        const { wrapper } = setupItemWithoutProductPageUrl()

        expect(wrapper.hasClass(style.btn + ' disabled')).toBe(true)
        expect(wrapper.prop("href")).toBeUndefined()
        expect(wrapper.find(FontAwesomeIcon).first().prop("icon")).toContain('external-link-square')
    })
})
