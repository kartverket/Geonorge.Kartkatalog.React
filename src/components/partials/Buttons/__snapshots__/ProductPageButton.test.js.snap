// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setupItemWithProductPageUrl should render self 1`] = `ShallowWrapper {}`;
